File: satyam-sundaram/monotonic_array.go
//package main

func isMonotonic(nums []int) bool {
    n := len(nums)
    if n == 1 { return true }
    increasing := true
    decreasing := true
    for i := 1; i < n; i++ {
        if increasing && nums[i] < nums[i-1] {
            increasing = false
        }
        if decreasing && nums[i] > nums[i-1] {
            decreasing = false
        }
    }
    return increasing || decreasing
}

File: Ravi-Teja/MonotonicArray.cpp
class Solution {
public:
    bool isMonotonic(vector<int>& nums) {
        bool increase=true, decrease=true;
        int n=nums.size();
        for(int i=0;i<n-1;i++){
            if(nums[i] < nums[i+1]){
                decrease=false;
            }
            else if(nums[i] > nums[i+1]){
                increase=false;
            }
            if(!increase & !decrease){
                return false;
            }
        }
        return increase || decrease;
    }
};